hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/DeadNodeTracker.java

```diff
@@ -0,0 +1,75 @@
+package org.apache.mesos.hdfs.state;
+
+import org.apache.commons.lang.time.DateUtils;
+import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.Date;
+
+/**
+ *
+ */
+public class DeadNodeTracker {
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
+  private Timestamp deadJournalNodeTimeStamp = null;
+  private Timestamp deadNameNodeTimeStamp = null;
+  private Timestamp deadDataNodeTimeStamp = null;
+
```

Comment:
We can remove this dependency now by passing the number of journal nodes, name nodes, and data nodes are parameters. I agree with you that it is cleaner code to remove that dependency. I also like that you broke this piece of logic out.